src/App.tsx

```diff
@@ -11,29 +12,73 @@ import { OnboardContext } from "common/context/OnboardContext";
 import { recoverPublicKey } from "./features/vote/helpers/recoverPublicKey";
 import { derivePrivateKey } from "./features/vote/helpers/derivePrivateKey";
 
+import currentSigningMessage from "common/currentSigningMessage";
+import { useCurrentRoundId } from "hooks";
+
 interface Props {
   queryClient: QueryClient;
 }
 
-export interface SigningKeys {
+export interface SigningKey {
   [key: string]: {
     publicKey: string;
     privateKey: string;
   };
 }
 
+export interface SigningKeys {
+  [key: string]: SigningKey;
+}
+
+const SIGNING_KEYS_STORAGE_KEY = "signingKeys";
+
 function App(props: Props) {
   const [signingKeys, setSigningKeys] = useState<SigningKeys>({});
 
   const { state, disconnect, dispatch } = useContext(OnboardContext);
   const { error, removeError, addError } = useContext(ErrorContext);
+  const { data: currentRoundId } = useCurrentRoundId();
 
   useEffect(() => {
     if (state.signer && state.address) {
       const address = state.address;
-      const message = "Login to UMA Voter dApp";
-      const keyExists = signingKeys[address];
-      if (!keyExists) {
+      const message = currentSigningMessage(Number(currentRoundId));
+      const hashedMessage = ethers.utils.formatBytes32String(message);
```

Comment:
Does formatBytes32String perform a hash? According to the docs, it doesn't seem to? It appears as if this is just taking a utf8 string and converting it to bytes32? 

Also, are we assuming the message won't be longer than 32 bytes?

Reply:
Fair, I can use a web3 func instead.